metadata/pkg/drivers/aws/aws.go

```diff
@@ -43,10 +50,122 @@ type AwsAdapter struct {
 	Session *session.Session
 }
 
+func iterateObjects(i int, object *s3.Object, svc *s3.S3, bucketName string, objectArray []model.MetaObject, wg *sync.WaitGroup) {
+	defer wg.Done()
+	obj := model.MetaObject{}
+	obj.LastModifiedDate = *object.LastModified
+	obj.ObjectName = *object.Key
+	obj.Size = *object.Size
+	obj.StorageClass = *object.StorageClass
+
+	meta, err := svc.HeadObject(&s3.HeadObjectInput{Bucket: &bucketName, Key: object.Key})
+	if err != nil {
+		log.Errorf("cannot perform head object on object %v in bucket %v. failed with error: %v\n", *object.Key, bucketName, err)
+	}
+	if meta.ServerSideEncryption != nil {
+		obj.ServerSideEncryption = *meta.ServerSideEncryption
+	}
+	if meta.VersionId != nil {
+		obj.VersionId = *meta.VersionId
+	}
+	obj.ObjectType = *meta.ContentType
+	if meta.Expires != nil {
+		expiresTime, err := time.Parse(time.RFC3339, *meta.Expires)
+		if err != nil {
+			log.Errorf("unable to parse given string to time type. error: %v. skipping ExpiresDate field\n", err)
+		} else {
+			obj.ExpiresDate = expiresTime
+		}
+	}
+	if meta.ReplicationStatus != nil {
+		obj.ReplicationStatus = *meta.ReplicationStatus
+	}
+	objectArray[i] = obj
+}
+
+func ObjectList(sess *session.Session, bucket *model.MetaBucket) error {
+	svc := s3.New(sess, aws.NewConfig().WithRegion(bucket.Region))
+	output, err := svc.ListObjectsV2(&s3.ListObjectsV2Input{Bucket: &bucket.Name})
+	if err != nil {
+		log.Errorf("unable to list objects in bucket %v. failed with error: %v\n", bucket.Name, err)
+		return err
+	}
+
+	numObjects := len(output.Contents)
+
+	objectArray := make([]model.MetaObject, numObjects)
+	wg := sync.WaitGroup{}
+	for i, object := range output.Contents {
+		wg.Add(1)
+		go iterateObjects(i, object, svc, bucket.Name, objectArray, &wg)
+	}
+	wg.Wait()
+	bucket.NumberOfObjects = numObjects
+	bucket.Objects = objectArray
+	return nil
+}
+
+func iterateBuckets(i int, bucket *s3.Bucket, sess *session.Session, bucketArray []model.MetaBucket, wg *sync.WaitGroup) {
+	defer wg.Done()
+	buck := model.MetaBucket{}
+	buck.CreationDate = *bucket.CreationDate
+	buck.Name = *bucket.Name
+	svc := s3.New(sess)
+	loc, err := svc.GetBucketLocation(&s3.GetBucketLocationInput{Bucket: bucket.Name})
+	if err != nil {
+		log.Errorf("unable to get bucket location. failed with error: %v\n", err)
+	} else {
+		buck.Region = *loc.LocationConstraint
+	}
+	newSvc := s3.New(sess, aws.NewConfig().WithRegion(buck.Region))
```

Comment:
you may ustilize previous session with region. Please check

Reply:
done